src/modules/RA/DataLayer/components/tabs/PopupTab/AddPopupField.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useCallback } from 'react';
+
+import Button from '@material-ui/core/Button';
+import Modal from '@material-ui/core/Modal';
+import Select from '@material-ui/core/Select';
+import  AddIcon from '@material-ui/icons/Add';
+import CancelIcon from '@material-ui/icons/Cancel';
+
+import { useTranslate } from 'react-admin';
+import { useForm, useField } from 'react-final-form';
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles({
+  modal: {
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    position: 'absolute',
+    width: 500,
+    backgroundColor: 'white',
+    padding: '1em',
+  },
+  modalButtons: {
+    marginTop: '1em',
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  fieldSelect: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+});
+
+const AddPopupField = ({ fields }) => {
```

Comment:
I think this is a similar code to the addFilter component. Should be refactored to a unique "SelectField" component and use "onSelect" callback to handle difference. What do you think ?

Reply:
It's a cleaner solution indeed.

Reply:
It has been refactored in `AddField.js`